fabric-common/lib/Utils.js

```diff
@@ -447,26 +447,15 @@ module.exports.getDefaultKeyStorePath = () => {
 	return path.join(os.homedir(), '.hfc-key-store');
 };
 
-const CryptoKeyStore = function (KVSImplClass, opts) {
-	let store;
-
-	this._getKeyStore = async function () {
-		if (!store) {
-			const newInstance = require('./impl/CryptoKeyStore');
-			store = await newInstance(KVSImplClass, opts);
-			await store.initialize();
-		}
-
-		return store;
-	};
-};
-
-module.exports.newCryptoKeyStore = (KVSImplClass, opts) => {
-	// this function supports skipping any of the arguments such that it can be called in any of the following fashions:
-	// - newCryptoKeyStore(CouchDBKeyValueStore, {name: 'member_db', url: 'http://localhost:5984'})
-	// - newCryptoKeyStore({path: '/tmp/app-state-store'})
-	// - newCryptoKeyStore()
-	return new CryptoKeyStore(KVSImplClass, opts);
+/**
+ *
+ * @param {KeyValueStore} [keyValueStore] Optional. The built-in key store saves private keys.
+ *    The key store must be instance of any {@link KeyValueStore} implementations.
+ * @return {CryptoKeyStore}
+ */
+module.exports.newCryptoKeyStore = (keyValueStore = new InMemoryKeyValueStore()) => {
+	const CryptoKeyStore = require('./impl/CryptoKeyStore');
```

Comment:
I guess it probably doesn't matter but do we need to do the require here or can it move to the top with the rest now?

Reply:
Accept